src/libs/memoize/types.ts

```diff
@@ -0,0 +1,45 @@
+import type {MemoizeStats} from './stats';
+
+type KeyComparator = <K>(key1: K[], key2: K[]) => boolean;
+
+type ValueBox<V> = {value: V};
+
+type Cache<K, V> = {
+    get: (key: K) => ValueBox<V> | undefined;
+    set: (key: K, value: V) => void;
+    getSet: (key: K, valueProducer: () => V) => ValueBox<V>;
```

Comment:
Can we add a code comment to explain what `getSet` does and when you might use it?

Reply:
Done ✅